features/cryptocell/FEATURE_CRYPTOCELL310/cmac_alt.h

```diff
@@ -0,0 +1,50 @@
+/*
+ *  cmac_alt.h
+ *
+ *  Copyright (C) 2019, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#ifndef __CMAC_ALT__
+#define __CMAC_ALT__
+
+#if defined(MBEDTLS_CMAC_ALT)
+#include "ssi_aes.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct mbedtls_cmac_context_t
```

Comment:
Do we need to typedef this so it works with C? It looks like perhaps this header was only tested with C++ builds.

Reply:
This was tested with our on target tests, which I believe is in c

Reply:
Mbed OS always builds with C++, so it's always C++ for on-target testing. We should use typedef here if we want to be C compatible. Otherwise, when you later refer to `mbedtls_cmac_context_t blah;` instead of `struct mbedtls_cmac_context_t blah`, you'll get compilation failures. I see many places in this PR where we don't say `struct mbedtls_cmac_context_t`.

Reply:
sure, I'll change but this implementation is for Mbed OS anyway